src/core/bob_client.rs

```diff
@@ -67,11 +67,14 @@ impl BobClient {
             }
         })
         .map_err(|e| {
-            println!("ERR = {:?}", e);
+            debug!("BobClient: ERR = {:?}", e);
         })
     }
 
-    pub fn put(&mut self, key: BobKey, data: &BobData) -> impl Future<Item=BobPutResult, Error=BobError> {
+    pub fn put(&mut self, key: BobKey, data: &BobData)
+            -> impl Future<Item=ClusterResult<BobPutResult>, Error=ClusterResult<BobError>> {
+        let n1 = self.node.clone();
+        let n2 = self.node.clone();
```

Comment:
Why do you clone it twice in the beginning of method?

Reply:
as below we need this variable in two closures. In rust they prefer explicit cloning